pointcloud/basic.go

```diff
@@ -51,3 +53,19 @@ func (cloud *basicPointCloud) Set(p r3.Vector, d Data) error {
 func (cloud *basicPointCloud) Iterate(numBatches, myBatch int, fn func(p r3.Vector, d Data) bool) {
 	cloud.points.Iterate(numBatches, myBatch, fn)
 }
+
+// vectorConvert converts a list of r3.Vectors into a pointcloud with the specified color
+func vectorConvert(vectors []r3.Vector, c color.NRGBA) (basicPointCloud, error) {
+	// initialize empty pointcloud
+	myCloud := basicPointCloud{
+		points: &matrixStorage{points: make([]PointAndData, 0, len(vectors)), indexMap: make(map[r3.Vector]uint, len(vectors))},
+		meta:   NewMetaData(),
+	}
+	// iterate throught the vector list and add to the pointcloud
+	for _, v := range vectors {
```

Comment:
should this be made concurrent?

Reply:
Same as other comment; add a TODO but let's not make it concurrent unless it's clear this is a perf issue.

Reply:
Likely this particular piece can't easily be made concurrent since it's modifying a struct in a not-thread-safe way